client/my-sites/checkout/composite-checkout/lib/get-jetpack-product-features.ts

```diff
@@ -0,0 +1,20 @@
+import { isJetpackBackupT1 } from '@automattic/calypso-products';
+import { useTranslate } from 'i18n-calypso';
+import type { WithSnakeCaseSlug, WithCamelCaseSlug } from '@automattic/calypso-products';
+
+export default function getJetpackProductFeatures(
+	product: WithSnakeCaseSlug | WithCamelCaseSlug,
+	translate: ReturnType< typeof useTranslate >
+): string[] {
+	if ( isJetpackBackupT1( product ) ) {
+		return [
+			String( translate( 'Real-time cloud backups' ) ),
```

Comment:
`translate()` actually returns `TranslateResult`, since the function can sometimes produce a React element if it uses interpolation. Obviously this string is not interpolated, but if you want TypeScript to know this you have to explicitly tell it by passing `{ textOnly: true }` like here:

https://github.com/Automattic/wp-calypso/blob/d05dbbe45e09c112baad817541945124f7e6555f/client/components/jetpack/profile-dropdown/index.tsx#L65

...or, you need to cast it to a string.

Reply:
Good to know the reasoning behind this! I will leave these casting as is then

Reply:
Ah, I see @sirbrillig; thank you for specifying that!

However, I meant that `TranslateResult` is a union type that includes `string`. That way, we don't need to force cast the value to string, but setting a return type for the function or variable (e.g. `const a: string = translate( ... )`) will already be correct from the perspective of Typescript code.

Sorry for making my comment confusing about that. I meant we cast here twice.

---

But I'll leave it as a suggestion, not a blocker.